src/dombeck_lab_to_nwb/azcorra2023/photometry_utils/add_fiber_photometry.py

```diff
@@ -0,0 +1,176 @@
+from collections import defaultdict
+from typing import Literal
+
+import numpy as np
+from hdmf.common import DynamicTableRegion
+from neuroconv.tools import get_module
+from pynwb import NWBFile
+from ndx_fiber_photometry import (
+    FiberPhotometryTable,
+    OpticalFiber,
+    Indicator,
+    ExcitationSource,
+    Photodetector,
+    DichroicMirror,
+    OpticalFilter,
+    FiberPhotometryResponseSeries,
+    CommandedVoltageSeries,
+)
+
+
+def add_photometry_device(nwbfile: NWBFile, device_metadata: dict, device_type: str):
+    """Add a photometry device to an NWBFile object."""
+
+    device_name = device_metadata["name"]
+    if device_name in nwbfile.devices:
+        return
+
+    photometry_device = defaultdict(
+        OpticalFiber=OpticalFiber,
+        Indicator=Indicator,
+        ExcitationSource=ExcitationSource,
+        Photodetector=Photodetector,
+        DichroicMirror=DichroicMirror,
+        OpticalFilter=OpticalFilter,
+    )[device_type](**device_metadata)
+
+    nwbfile.add_device(photometry_device)
```

Comment:
@pauladkisson @luiztauffer This has really simplified creating the "devices" from the metadata for me. Maybe will be useful also for TDT fiber photometry interface on `neuroconv`?

Reply:
Thanks szonja! I'll try it out.